src/native/libs/System.Native/pal_time.c

```diff
@@ -95,22 +95,22 @@ int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
     uint64_t lastRecordedKernelTime = previousCpuInfo->lastRecordedKernelTime;
     uint64_t lastRecordedUserTime = previousCpuInfo->lastRecordedUserTime;
 
-    uint64_t cpuTotalTime = 0;
+    double cpuTotalTime = 0.0;
     if (currentTime > lastRecordedCurrentTime)
     {
         cpuTotalTime = (currentTime - lastRecordedCurrentTime);
     }
 
-    uint64_t cpuBusyTime = 0;
+    double cpuBusyTime = 0.0;
     if (userTime >= lastRecordedUserTime && kernelTime >= lastRecordedKernelTime)
     {
         cpuBusyTime = (userTime - lastRecordedUserTime) + (kernelTime - lastRecordedKernelTime);
     }
 
-    int32_t cpuUtilization = 0;
+    double cpuUtilization = 0.0;
```

Comment:
the method return type needs to be changed to double, otherwise cpuUtilization will be rounded back to an integer when it is returned the caller.